analyzer/src/main/kotlin/managers/Yarn2.kt

```diff
@@ -126,21 +181,503 @@ class Yarn2(
         // We depend on a version >= 2, so we check the version for safety.
         definitionFiles.forEach { checkVersion(it.parentFile) }
 
-    override fun runInstall(workingDir: File): ProcessCapture =
-        // It should be enough for Yarn2+ to parse the lockfile to get the dependencies, and, if the former doesn't
-        // exist, run `yarn install --mode update-lockfile` to generate it (since yarn version 3.0.0-rc.10).
-        // Unfortunately, the lockfile doesn't contain the information required to build ORT's packages. An alternative
-        // could be to run `npm show --json` on each package present in the lockfile and rebuild a node_modules
-        // file hierarchy to be able to reuse the code present in the Npm package manager.
-        // Fortunately, an easier solution exist: by setting the option `nodeLinker` in the `yarnrc` file or as an
-        // environment variable, one can instruct Yarn to generate the node_modules files hierarchy.
-        run("install", workingDir = workingDir, environment = mapOf("YARN_NODE_LINKER" to "node-modules"))
-
-    override fun getRemotePackageDetails(workingDir: File, packageName: String): JsonNode {
-        // Yarn2+ 'info' command does not provide enough information so the default behavior of 'npm view --json' is
-        // kept. Note that it is mandatory to override this function as the super function calls `run()` which would
-        // call Yarn.
-        val process = delegateNpmAnalyzer.run(workingDir, "view", "--json", packageName)
-        return jsonMapper.readTree(process.stdout)
+    override fun resolveDependencies(definitionFile: File, labels: Map<String, String>): List<ProjectAnalyzerResult> {
+        val workingDir = definitionFile.parentFile
+
+        // Running `yarn install` before `yarn info` allows to get the real local package version. Otherwise, it will be
+        // a generic one such as '0.0.0-use.local'.
+        log.info { "Running 'Yarn install'..." }
+
+        run("install", workingDir = workingDir)
+
+        log.info { "Running 'Yarn info'..." }
```

Comment:
`run` already logs the commands that are run on `debug`. I think you can remove the info logs.

Reply:
Well, we go again in the debate we had for `FossID` scanner :)

I think a software running several steps after one another should log these steps, so that the user doesn't think the application is hanging.
Naturally, this logging should be sporadic and "high level". For instance here each package requested with `npm list` is NOT logged.
If desired, this is a task for the `DEBUG` level.

Reply:
Yes, but as of now, the info log prints duplicated information:
```
14:57:32.293 [DefaultDispatcher-worker-1] INFO  org.ossreviewtoolkit.analyzer.managers.Yarn2 - Resolving Yarn2 dependencies for path 'package.json'...
14:57:32.295 [DefaultDispatcher-worker-1] INFO  org.ossreviewtoolkit.analyzer.managers.Yarn2 - Running 'Yarn install'...
14:57:32.299 [DefaultDispatcher-worker-1] INFO  org.ossreviewtoolkit.utils.common.ProcessCapture - Running '/Users/bom1wa3/work/playground/yarn-V2-workspaces-simple-monorepo/.yarn/releases/yarn-2.2.2.cjs install' in '/Users/bom1wa3/work/playground/yarn-V2-workspaces-simple-monorepo'...
14:57:58.218 [DefaultDispatcher-worker-1] INFO  org.ossreviewtoolkit.analyzer.managers.Yarn2 - Running 'Yarn info'...
14:57:58.223 [DefaultDispatcher-worker-1] INFO  org.ossreviewtoolkit.utils.common.ProcessCapture - Running '/Users/bom1wa3/work/playground/yarn-V2-workspaces-simple-monorepo/.yarn/releases/yarn-2.2.2.cjs info -A -R --manifest --json' in '/Users/bom1wa3/work/playground/yarn-V2-workspaces-simple-monorepo'...
```

Reply:
You have a point I removed them.